On update apply, robocopy fallback fails, leaving app in unlaunchable state

### OS

Windows 10.0.22631

### Programming Language

net462

### VPK Version

0.0.589-g4087722

### Library Version

Nuget 0.0.589-g4087722

### What happened?

I have gathered some diagnostic information from affected customers reporting that the app "spins and doesn't actually launch". Inspecting the WER logs in Event Viewer reveals a `FileNotFoundException` in the entry point of the app, indicating that one or more DLLs are missing during assembly loading.

The root of this is an "access denied" error preventing the updater from renaming the `current` folder. ~~As part of the app, there is a second executable that is launched and communicated with via IPC. I suspect that this second executable is not closing due to a bug I have introduced recently, keeping the folder locked.~~

**Update 2024-09-19**:
Another customer WER log reveals this exception, possibly suggesting two updates happening at once.

```
Category: Velopack.UpdateManager
EventId: 0

Unable to apply delta updates, falling back to full update.

Exception:
System.IO.IOException: The process cannot access the file 'C:\Users\username\AppData\Local\AppName\packages\VelopackTemp\temp.1\lib\app\AppName.Services.dll' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.IO.File.OpenFile(String path, FileAccess access, SafeFileHandle& handle)
   at System.IO.File.SetLastWriteTimeUtc(String path, DateTime lastWriteTimeUtc)
   at System.IO.Compression.ZipFileExtensions.ExtractToFile(ZipArchiveEntry source, String destinationFileName, Boolean overwrite)
   at Velopack.Compression.EasyZip.ExtractRelativeToDirectory(ZipArchiveEntry source, String destinationDirectoryName, Boolean overwrite, Boolean expandSymlinks)
   at Velopack.Compression.EasyZip.ExtractZipToDirectory(ILogger logger, String inputFile, String outputDirectory, Boolean expandSymlinks)
   at Velopack.UpdateManager.<DownloadUpdatesAsync>d__35.MoveNext()
```

This is despite the use of a mutex: https://github.com/velopack/velopack/blob/b828850b45fd50401a90c0855693c01fa6edca36/src/lib-csharp/UpdateManager.cs#L229

The velopack.log is essentially identical to the one provided below.

### Relevant log output

```shell
[update:15776] [08:54:24] [INFO] --
[update:15776] [08:54:24] [INFO] Starting Velopack Updater (0.0.589-g4087722)
[update:15776] [08:54:24] [INFO]     Location: C:\Users\username\AppData\Local\AppName\Update.exe
[update:15776] [08:54:24] [INFO]     CWD: C:\Users\username\AppData\Local\AppName
[update:15776] [08:54:24] [INFO]     Verbose: false
[update:15776] [08:54:24] [INFO]     Silent: false
[update:15776] [08:54:24] [INFO]     Log File: None
[update:15776] [08:54:24] [INFO] Command: Apply
[update:15776] [08:54:24] [INFO]     Restart: true
[update:15776] [08:54:24] [INFO]     Wait: WaitPid(10972)
[update:15776] [08:54:24] [INFO]     Package: Some("C:\\Users\\username\\AppData\\Local\\AppName\\packages\\AppName-1.9023.0-win10.0.14393-full.nupkg")
[update:15776] [08:54:24] [INFO]     Exe Args: None
[update:15776] [08:54:24] [INFO] Loaded manifest for application: AppName
[update:15776] [08:54:24] [INFO] Root Directory: C:\Users\username\AppData\Local\AppName
[update:15776] [08:54:24] [INFO] Attempting to open global system mutex: 'velopack-AppName'
[update:15776] [08:54:24] [INFO] Waiting 60000ms for process (10972) to exit.
[update:15776] [08:54:24] [INFO] Getting ready to apply package to AppName ver 1.9018.0: C:\Users\username\AppData\Local\AppName\packages\AppName-1.9023.0-win10.0.14393-full.nupkg
[update:15776] [08:54:24] [INFO] Applying package to current: 1.9023.0 (old version 1.9018.0)
[update:15776] [08:54:24] [INFO] Checking application pre-requisites...
[update:15776] [08:54:24] [INFO]     .NET Framework 4.6.2 is already installed.
[update:15776] [08:54:24] [INFO] Extracting 240 app files to 'C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y'...
[update:15776] [08:54:25] [INFO] Running --veloapp-obsolete hook...
[update:15776] [08:54:26] [INFO] Hook executed successfully (took 123.582ms)
[update:15776] [08:54:26] [INFO] Checking for running processes in: C:\Users\username\AppData\Local\AppName
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\locationnotificationwindows.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\securityhealthsystray.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\applicationframehost.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\oobe\useroobebroker.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Skipping killing self: C:\Users\username\AppData\Local\AppName\Update.exe (15776)
[update:15776] [08:54:26] [INFO] Checking for running processes in: C:\Users\username\AppData\Local\AppName
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\locationnotificationwindows.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\securityhealthsystray.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\applicationframehost.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Failed to get long path for 'c:\windows\system32\oobe\useroobebroker.exe': The system cannot find the file specified. (0x80070002)
[update:15776] [08:54:26] [WARN] Skipping killing self: C:\Users\username\AppData\Local\AppName\Update.exe (15776)
[update:15776] [08:54:27] [INFO] Backing up current dir to C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1
[update:15776] [08:54:27] [INFO] Replacing current dir with C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y
[update:15776] [08:54:27] [WARN] Failed to rename temp_path_new to current_dir (Access is denied. (os error 5)). Retrying with robocopy...
[update:15776] [08:54:34] [ERROR] 
-------------------------------------------------------------------------------
   ROBOCOPY     ::     Robust File Copy for Windows                              
-------------------------------------------------------------------------------

  Started : Monday, September 16, 2024 8:54:28 AM
   Source : C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\
     Dest : C:\Users\username\AppData\Local\AppName\current\

    Files : *.*
	    
  Options : *.* /S /E /DCOPY:DA /COPY:DAT /PURGE /MIR /IS /MT:2 /R:5 /W:1 

------------------------------------------------------------------------------

	    New File  		  209488	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\CommandLine.dll
	    New File  		  143440	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\DataGridPrimary.dll
	    New File  		  693328	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\DynamicData.dll
	    New File  		   1.2 m	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\EPPlus.dll
	    New File  		   21072	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Fastenshtein.dll
	    New File  		  256952	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Humanizer.dll
	    New File  		   70736	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Karambolo.Extensions.Logging.File.dll
	    New File  		   26904	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Bcl.AsyncInterfaces.dll
	    New File  		  135248	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Data.Sqlite.dll
	    New File  		   24536	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.DotNet.PlatformAbstractions.dll
	    New File  		   20560	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.EntityFrameworkCore.Abstractions.dll
	    New File  		   1.4 m	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.EntityFrameworkCore.dll
	    New File  		  788856	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.EntityFrameworkCore.Relational.dll
	    New File  		  125304	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.EntityFrameworkCore.Sqlite.dll
	    New File  		  108168	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Expression.Interactions.dll
	    New File  		   26688	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Caching.Abstractions.dll
	    New File  		   32312	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Caching.Memory.dll
	    New File  		   31392	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.Abstractions.dll
	    New File  		   48400	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.Binder.dll
	    New File  		   28424	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.CommandLine.dll
	    New File  		   47376	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.dll
	    New File  		   24856	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.EnvironmentVariables.dll
	    New File  		   31496	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.FileExtensions.dll
	    New File  		   30368	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.Json.dll
	    New File  		   28944	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Configuration.UserSecrets.dll
	    New File  		   64160	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.DependencyInjection.Abstractions.dll
	    New File  		   96928	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.DependencyInjection.dll
	    New File  		   62936	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.DependencyModel.dll
	    New File  		   34568	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Diagnostics.Abstractions.dll
	    New File  		   38672	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Diagnostics.dll
	    New File  		   25888	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.FileProviders.Abstractions.dll
	    New File  		   50448	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.FileProviders.Physical.dll
	    New File  		   49440	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.FileSystemGlobbing.dll
	    New File  		   55576	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Hosting.Abstractions.dll
	    New File  		   79008	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Hosting.dll
	    New File  		   80152	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Http.dll
	    New File  		   67848	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.Abstractions.dll
	    New File  		   32008	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.Configuration.dll
	    New File  		   79120	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.Console.dll
	    New File  		   23840	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.Debug.dll
	    New File  		   54032	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.dll
	    New File  		   30480	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.EventLog.dll
	    New File  		   40224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Logging.EventSource.dll
	    New File  		   27920	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Options.ConfigurationExtensions.dll
	    New File  		   68768	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Options.dll
	    New File  		   49936	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Extensions.Primitives.dll
	    New File  		   21216	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Microsoft.Win32.Primitives.dll
	    New File  		   98616	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\netstandard.dll
	    New File  		  701992	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Newtonsoft.Json.dll
	    New File  		   68544	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\NuGet.Versioning.dll
	    New File  		  121288	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Ookii.Dialogs.Wpf.dll
	    New File  		  527816	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\PdfSharp-WPF.dll
	    New File  		  142280	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\PdfSharp.Xps.dll
	    New File  		  239536	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\ReactiveUI.dll
	    New File  		   46000	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\ReactiveUI.WPF.dll
	    New File  		  197712	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Remotion.Linq.dll
	    New File  		   29120	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\securifybv.PropertyStore.dll
	    New File  		   95688	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\securifybv.ShellLink.dll
	    New File  		   46024	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\SimpleBase.dll
	    New File  		   52656	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Splat.dll
	    New File  		     633	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\sq.version
	    New File  		   20048	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\SQLitePCLRaw.batteries_green.dll
	    New File  		   20048	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\SQLitePCLRaw.batteries_v2.dll
	    New File  		   53328	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\SQLitePCLRaw.core.dll
	    New File  		   54864	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\SQLitePCLRaw.provider.e_sqlite3.dll
	    New File  		   1.2 m	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\AppName.exe
	    New File  		   26690	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\AppName.exe.config
	    New File  		   32336	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\AppName.Licensing.dll
	    New File  		   21176	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.AppContext.dll
	    New File  		   20856	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Buffers.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Collections.Concurrent.dll
	    New File  		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Collections.dll
	    New File  		  198784	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Collections.Immutable.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Collections.NonGeneric.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Collections.Specialized.dll
	    New File  		   43152	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ComponentModel.Annotations.dll
	    New File  		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ComponentModel.dll
	    New File  		   21768	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ComponentModel.EventBasedAsync.dll
	    New File  		   21752	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ComponentModel.Primitives.dll
	    New File  		   22784	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ComponentModel.TypeConverter.dll
	    New File  		   21160	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Console.dll
	    New File  		  154448	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Data.Common.dll
	    New File  		   21736	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.Contracts.dll
	    New File  		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.Debug.dll
	    New File  		  189088	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.DiagnosticSource.dll
	    New File  		   21248	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.FileVersionInfo.dll
	    New File  		   21728	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.Process.dll
	    New File  		   23936	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.StackTrace.dll
	    New File  		   21280	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.TextWriterTraceListener.dll
	    New File  		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.Tools.dll
	    New File  		   21744	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.TraceSource.dll
	    New File  		   31608	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Diagnostics.Tracing.dll
	    New File  		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Drawing.Primitives.dll
	    New File  		   22224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Dynamic.Runtime.dll
	    New File  		   21744	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Globalization.Calendars.dll
	    New File  		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Globalization.dll
	    New File  		   25992	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Globalization.Extensions.dll
	    New File  		  252344	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Interactive.Async.dll
	    New File  		  110944	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.Compression.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.Compression.ZipFile.dll
	    New File  		   21144	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.dll
	    New File  		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.FileSystem.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.FileSystem.DriveInfo.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.FileSystem.Primitives.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.FileSystem.Watcher.dll
	    New File  		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.IsolatedStorage.dll
	    New File  		   21216	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.MemoryMappedFiles.dll
	    New File  		   21168	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.Pipes.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.IO.UnmanagedMemoryStream.dll
	    New File  		   21152	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Linq.dll
	    New File  		   22224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Linq.Expressions.dll
	    New File  		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Linq.Parallel.dll
	    New File  		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Linq.Queryable.dll
	    New File  		  142240	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Memory.dll
	    New File  		  198472	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Http.dll
	    New File  		   72864	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Http.Json.dll
	    New File  		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.NameResolution.dll
	    New File  		   22248	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.NetworkInformation.dll
	    New File  		   21168	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Ping.dll
	    New File  		   22216	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Primitives.dll
	    New File  		   21184	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Requests.dll
	    New File  		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Security.dll
	    New File  		   30544	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.Sockets.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.WebHeaderCollection.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.WebSockets.Client.dll
	    New File  		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Net.WebSockets.dll
	    New File  		  115856	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Numerics.Vectors.dll
	    New File  		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ObjectModel.dll
	    New File  		   1.2 m	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Reactive.dll
	    New File  		   22200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Reflection.dll
	    New File  		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Reflection.Extensions.dll
	    New File  		   21736	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Reflection.Primitives.dll
	    New File  		   80544	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Resources.Extensions.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Resources.Reader.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Resources.ResourceManager.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Resources.Writer.dll
	    New File  		   18024	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.CompilerServices.Unsafe.dll
	    New File  		   21784	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.CompilerServices.VisualC.dll
	    New File  		   29360	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.dll
	    New File  		   21720	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Extensions.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Handles.dll
	    New File  		   24296	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.InteropServices.dll
	    New File  		   28624	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.InteropServices.RuntimeInformation.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Numerics.dll
	    New File  		   21264	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Serialization.Formatters.dll
	    New File  		   21240	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Serialization.Json.dll
	    New File  		   27048	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Serialization.Primitives.dll
	    New File  		   24816	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Runtime.Serialization.Xml.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Claims.dll
	    New File  		   47016	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Cryptography.Algorithms.dll
	    New File  		   21240	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Cryptography.Csp.dll
	    New File  		   21256	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Cryptography.Encoding.dll
	    New File  		   21784	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Cryptography.Primitives.dll
	    New File  		   22832	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Cryptography.X509Certificates.dll
	    New File  		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.Principal.dll
	    New File  		   22392	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Security.SecureString.dll
	    New File  		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Text.Encoding.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Text.Encoding.Extensions.dll
	    New File  		   79024	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Text.Encodings.Web.dll
	    New File  		  643864	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Text.Json.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Text.RegularExpressions.dll
	    New File  		   45952	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Channels.dll
	    New File  		   22192	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.dll
	    New File  		   37752	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Overlapped.dll
	    New File  		   22224	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Tasks.dll
	    New File  		   25984	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Tasks.Extensions.dll
	    New File  		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Tasks.Parallel.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Thread.dll
	    New File  		   21216	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.ThreadPool.dll
	    New File  		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Threading.Timer.dll
	    New File  		   78992	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.ValueTuple.dll
	    New File  		   55904	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Windows.Interactivity.dll
	    New File  		   21712	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Xml.ReaderWriter.dll
	    New File  		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Xml.XDocument.dll
	    New File  		   21712	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Xml.XmlDocument.dll
	    New File  		   21712	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Xml.XmlSerializer.dll
	    New File  		   21176	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Xml.XPath.dll
	    New File  		   22904	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\System.Xml.XPath.XDocument.dll
	    New File  		   61384	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\Trestat.MsixShortcut.dll
	    New File  		  857544	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\USCRAM.Data.dll
	    New File  		  100944	C:\Users\username\AppData\Local\AppName\packages\tmp_vheL7I6ye3cGyZ3y\USCRAM.Documentation.dll
[update:15776] [08:54:34] [ERROR] 
[update:15776] [08:54:34] [ERROR] Failed to robocopy temp_path_new to current_dir (Robocopy failed with code: -1073741510). Will attempt a rollback...
[update:15776] [08:54:44] [ERROR] 
-------------------------------------------------------------------------------
   ROBOCOPY     ::     Robust File Copy for Windows                              
-------------------------------------------------------------------------------

  Started : Monday, September 16, 2024 8:54:34 AM
   Source : C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\
     Dest : C:\Users\username\AppData\Local\AppName\current\

    Files : *.*
	    
  Options : *.* /S /E /DCOPY:DA /COPY:DAT /PURGE /MIR /IS /MT:2 /R:5 /W:1 

------------------------------------------------------------------------------

	    Older     		  209488	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\CommandLine.dll
	    Older     		  144328	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\DataGridPrimary.dll
	    Older     		  694208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\DynamicData.dll
	    Older     		   1.2 m	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\EPPlus.dll
	    Older     		   21952	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Fastenshtein.dll
	    Older     		  256952	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Humanizer.dll
	    Older     		   71624	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Karambolo.Extensions.Logging.File.dll
	    Older     		   26904	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Bcl.AsyncInterfaces.dll
	    Older     		  135248	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Data.Sqlite.dll
	    Older     		   24536	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.DotNet.PlatformAbstractions.dll
	    Older     		   20560	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.EntityFrameworkCore.Abstractions.dll
	    Older     		   1.4 m	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.EntityFrameworkCore.dll
	    Older     		  788856	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.EntityFrameworkCore.Relational.dll
	    Older     		  125304	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.EntityFrameworkCore.Sqlite.dll
	    Older     		  108168	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Expression.Interactions.dll
	    Older     		   26688	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Caching.Abstractions.dll
	    Older     		   32312	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Caching.Memory.dll
	    Older     		   31392	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.Abstractions.dll
	    Older     		   48400	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.Binder.dll
	    Older     		   28424	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.CommandLine.dll
	    Older     		   47376	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.dll
	    Older     		   24856	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.EnvironmentVariables.dll
	    Older     		   31496	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.FileExtensions.dll
	    Older     		   30368	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.Json.dll
	    Older     		   28944	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Configuration.UserSecrets.dll
	    Older     		   64160	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.DependencyInjection.Abstractions.dll
	    Older     		   96928	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.DependencyInjection.dll
	    Older     		   62936	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.DependencyModel.dll
	    Older     		   34568	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Diagnostics.Abstractions.dll
	    Older     		   38672	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Diagnostics.dll
	    Older     		   25888	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.FileProviders.Abstractions.dll
	    Older     		   50448	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.FileProviders.Physical.dll
	    Older     		   49440	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.FileSystemGlobbing.dll
	    Older     		   55576	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Hosting.Abstractions.dll
	    Older     		   79008	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Hosting.dll
	    Older     		   80152	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Http.dll
	    Older     		   67848	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.Abstractions.dll
	    Older     		   32008	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.Configuration.dll
	    Older     		   79120	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.Console.dll
	    Older     		   23840	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.Debug.dll
	    Older     		   54032	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.dll
	    Older     		   30480	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.EventLog.dll
	    Older     		   40224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Logging.EventSource.dll
	    Older     		   27920	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Options.ConfigurationExtensions.dll
	    Older     		   68768	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Options.dll
	    Older     		   49936	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Extensions.Primitives.dll
	    Older     		   21216	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Microsoft.Win32.Primitives.dll
	    Older     		   98616	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\netstandard.dll
	    Older     		  701992	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Newtonsoft.Json.dll
	    Older     		   68544	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\NuGet.Versioning.dll
	    Older     		  121280	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Ookii.Dialogs.Wpf.dll
	    Older     		  527816	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\PdfSharp-WPF.dll
	    Older     		  142280	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\PdfSharp.Xps.dll
	    Older     		  239536	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\ReactiveUI.dll
	    Older     		   46000	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\ReactiveUI.WPF.dll
	    Older     		  198600	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Remotion.Linq.dll
	    Older     		   29120	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\securifybv.PropertyStore.dll
	    Older     		   95680	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\securifybv.ShellLink.dll
	    Older     		   46016	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\SimpleBase.dll
	    Older     		   52656	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Splat.dll
	    Older     		     633	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\sq.version
	    Older     		   20936	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\SQLitePCLRaw.batteries_green.dll
	    Older     		   20936	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\SQLitePCLRaw.batteries_v2.dll
	    Older     		   54216	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\SQLitePCLRaw.core.dll
	    Older     		   54864	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\SQLitePCLRaw.provider.e_sqlite3.dll
	    Older     		   1.2 m	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\AppName.exe
	    Older     		   26690	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\AppName.exe.config
	    Older     		   33224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\AppName.Licensing.dll
	    Older     		   21176	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.AppContext.dll
	    Older     		   20856	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Buffers.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Collections.Concurrent.dll
	    Older     		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Collections.dll
	    Older     		  198784	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Collections.Immutable.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Collections.NonGeneric.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Collections.Specialized.dll
	    Older     		   43152	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ComponentModel.Annotations.dll
	    Older     		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ComponentModel.dll
	    Older     		   21768	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ComponentModel.EventBasedAsync.dll
	    Older     		   21752	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ComponentModel.Primitives.dll
	    Older     		   22784	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ComponentModel.TypeConverter.dll
	    Older     		   21160	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Console.dll
	    Older     		  154448	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Data.Common.dll
	    Older     		   21736	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.Contracts.dll
	    Older     		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.Debug.dll
	    Older     		  189088	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.DiagnosticSource.dll
	    Older     		   21248	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.FileVersionInfo.dll
	    Older     		   21728	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.Process.dll
	    Older     		   23936	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.StackTrace.dll
	    Older     		   21280	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.TextWriterTraceListener.dll
	    Older     		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.Tools.dll
	    Older     		   21744	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.TraceSource.dll
	    Older     		   31608	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Diagnostics.Tracing.dll
	    Older     		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Drawing.Primitives.dll
	    Older     		   22224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Dynamic.Runtime.dll
	    Older     		   21744	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Globalization.Calendars.dll
	    Older     		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Globalization.dll
	    Older     		   25992	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Globalization.Extensions.dll
	    Older     		  252344	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Interactive.Async.dll
	    Older     		  110944	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.Compression.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.Compression.ZipFile.dll
	    Older     		   21144	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.dll
	    Older     		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.FileSystem.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.FileSystem.DriveInfo.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.FileSystem.Primitives.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.FileSystem.Watcher.dll
	    Older     		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.IsolatedStorage.dll
	    Older     		   21216	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.MemoryMappedFiles.dll
	    Older     		   21168	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.Pipes.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.IO.UnmanagedMemoryStream.dll
	    Older     		   21152	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Linq.dll
	    Older     		   22224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Linq.Expressions.dll
	    Older     		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Linq.Parallel.dll
	    Older     		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Linq.Queryable.dll
	    Older     		  142240	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Memory.dll
	    Older     		  198472	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Http.dll
	    Older     		   72864	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Http.Json.dll
	    Older     		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.NameResolution.dll
	    Older     		   22248	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.NetworkInformation.dll
	    Older     		   21168	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Ping.dll
	    Older     		   22216	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Primitives.dll
	    Older     		   21184	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Requests.dll
	    Older     		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Security.dll
	    Older     		   30544	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.Sockets.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.WebHeaderCollection.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.WebSockets.Client.dll
	    Older     		   21192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Net.WebSockets.dll
	    Older     		  115856	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Numerics.Vectors.dll
	    Older     		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ObjectModel.dll
	    Older     		   1.2 m	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Reactive.dll
	    Older     		   22200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Reflection.dll
	    Older     		   21224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Reflection.Extensions.dll
	    Older     		   21736	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Reflection.Primitives.dll
	    Older     		   80544	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Resources.Extensions.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Resources.Reader.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Resources.ResourceManager.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Resources.Writer.dll
	    Older     		   18024	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.CompilerServices.Unsafe.dll
	    Older     		   21784	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.CompilerServices.VisualC.dll
	    Older     		   29360	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.dll
	    Older     		   21720	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Extensions.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Handles.dll
	    Older     		   24296	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.InteropServices.dll
	    Older     		   28624	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.InteropServices.RuntimeInformation.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Numerics.dll
	    Older     		   21264	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Serialization.Formatters.dll
	    Older     		   21240	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Serialization.Json.dll
	    Older     		   27048	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Serialization.Primitives.dll
	    Older     		   24816	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Runtime.Serialization.Xml.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Claims.dll
	    Older     		   47016	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Cryptography.Algorithms.dll
	    Older     		   21240	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Cryptography.Csp.dll
	    Older     		   21256	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Cryptography.Encoding.dll
	    Older     		   21784	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Cryptography.Primitives.dll
	    Older     		   22832	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Cryptography.X509Certificates.dll
	    Older     		   21208	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.Principal.dll
	    Older     		   22392	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Security.SecureString.dll
	    Older     		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Text.Encoding.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Text.Encoding.Extensions.dll
	    Older     		   79024	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Text.Encodings.Web.dll
	    Older     		  643864	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Text.Json.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Text.RegularExpressions.dll
	    Older     		   45952	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Channels.dll
	    Older     		   22192	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.dll
	    Older     		   37752	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Overlapped.dll
	    Older     		   22224	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Tasks.dll
	    Older     		   25984	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Tasks.Extensions.dll
	    Older     		   21232	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Tasks.Parallel.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Thread.dll
	    Older     		   21216	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.ThreadPool.dll
	    Older     		   21200	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Threading.Timer.dll
	    Older     		   78992	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.ValueTuple.dll
	    Older     		   55904	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Windows.Interactivity.dll
	    Older     		   21712	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Xml.ReaderWriter.dll
	    Older     		   21696	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Xml.XDocument.dll
	    Older     		   21712	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Xml.XmlDocument.dll
	    Older     		   21712	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Xml.XmlSerializer.dll
	    Older     		   21176	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Xml.XPath.dll
	    Older     		   22904	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\System.Xml.XPath.XDocument.dll
	    Older     		   60496	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\Trestat.MsixShortcut.dll
	    Older     		  853072	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Data.dll
	    Older     		  100944	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documentation.dll
	    Older     		   39504	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Core.dll
	    New File  		   40904	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.A.Domain.dll
	    New File  		   38848	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.A.Query.dll
	    New File  		   52672	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.B.Domain.dll
	    New File  		   49600	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.B.Query.dll
	    New File  		   40384	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.C.Domain.dll
	    New File  		   38344	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.C.Query.dll
	    New File  		   49600	C:\Users\username\AppData\Local\AppName\packages\tmp_8197wZCVLUbXGny1\USCRAM.Documents.Reports.D.Domain.dll
[update:15776] [08:54:44] [ERROR] 
[update:15776] [08:54:44] [INFO] Showing error dialog...
[update:15776] [08:54:48] [ERROR] Error applying package: Fatal error performing update.
[update:15776] [08:54:48] [INFO] About to launch: 'C:\Users\username\AppData\Local\AppName\current\AppName.exe' in dir 'C:\Users\username\AppData\Local\AppName\current'
[update:15776] [08:54:48] [INFO] Args: CommandArgs { inner: [] }
[update:15776] [08:54:48] [ERROR] Apply error: Apply failed, see logs for details.
```

--- a/src/bins/Cargo.toml
+++ b/src/bins/Cargo.toml
@@ -105,7 +105,7 @@ webview2-com = "0.33"
 libloading.workspace = true
 strsim.workspace = true
 same-file.workspace = true
-filelocksmith.workspace = true
+# filelocksmith.workspace = true
 
 [dev-dependencies]
 tempfile.workspace = true

--- a/src/bins/src/commands/apply.rs
+++ b/src/bins/src/commands/apply.rs
@@ -39,19 +39,18 @@ pub fn apply<'a>(
                     return Ok(applied_locator);
                 }
                 Err(e) => {
-                    error!("Error applying package: {}", e);
+                    if restart {
+                        shared::start_package(&locator, exe_args, Some(constants::HOOK_ENV_RESTART))?;
+                    }
+                    bail!("Error applying package: {}", e);
                 }
             }
         }
         None => {
-            error!("Failed to locate full package to apply. Please provide with the --package {{path}} argument");
+            if restart {
+                shared::start_package(&locator, exe_args, Some(constants::HOOK_ENV_RESTART))?;
+            }
+            bail!("Failed to locate full package to apply. Please provide with the --package {{path}} argument");
         }
     }
-
-    // an error occurred if we're here, but we still want to restart the old version of the app if it was requested
-    if restart {
-        shared::start_package(&locator, exe_args, Some(constants::HOOK_ENV_RESTART))?;
-    }
-
-    bail!("Apply failed, see logs for details.");
 }

--- a/src/bins/src/commands/apply_windows_impl.rs
+++ b/src/bins/src/commands/apply_windows_impl.rs
@@ -6,10 +6,7 @@ use crate::{
 };
 use anyhow::{bail, Context, Result};
 use std::sync::mpsc;
-use std::{
-    fs,
-    path::{Path, PathBuf},
-};
+use std::{fs, path::PathBuf};
 use velopack::{bundle::load_bundle_from_file, constants, locator::VelopackLocator};
 
 // fn ropycopy<P1: AsRef<Path>, P2: AsRef<Path>>(source: &P1, dest: &P2) -> Result<()> {
@@ -94,7 +91,7 @@ pub fn apply_package_impl(old_locator: &VelopackLocator, package: &PathBuf, run_
         // fourth, we make as backup of the current dir to temp_path_old
         info!("Backing up current dir to {}", &temp_path_old.to_string_lossy());
         shared::retry_io_ex(|| fs::rename(&current_dir, &temp_path_old), 1000, 10)
-            .context("Unable to start the update, because one or more running processes prevented it.")?;
+            .context("Unable to start the update, because one or more running processes prevented it. Try again later, or if the issue persists, restart your computer.")?;
 
         // let mut requires_robocopy = false;
         // if let Err(e) = fs::rename(&current_dir, &temp_path_old) {
@@ -107,7 +104,7 @@ pub fn apply_package_impl(old_locator: &VelopackLocator, package: &PathBuf, run_
         // if this fails we will yolo a rollback...
         info!("Replacing current dir with {}", &temp_path_new.to_string_lossy());
         shared::retry_io_ex(|| fs::rename(&temp_path_new, &current_dir), 1000, 30)
-            .context("Unable to complete the update, and the app was left in a broken state. You may need to re-install")?;
+            .context("Unable to complete the update, and the app was left in a broken state. You may need to re-install or repair this application manually.")?;
 
         // if !requires_robocopy {
         //     // if we didn't need robocopy for the backup, we don't need it for the deploy hopefully

--- a/src/bins/src/commands/apply_windows_impl.rs
+++ b/src/bins/src/commands/apply_windows_impl.rs
@@ -1,46 +1,46 @@
 use crate::{
     dialogs,
     shared::{self},
-    windows::locksmith,
+    // windows::locksmith,
     windows::splash,
 };
-use anyhow::{bail, Result};
+use anyhow::{bail, Context, Result};
 use std::sync::mpsc;
 use std::{
     fs,
     path::{Path, PathBuf},
 };
-use velopack::{bundle::load_bundle_from_file, locator::VelopackLocator, constants};
+use velopack::{bundle::load_bundle_from_file, constants, locator::VelopackLocator};
 
-fn ropycopy<P1: AsRef<Path>, P2: AsRef<Path>>(source: &P1, dest: &P2) -> Result<()> {
-    let source = source.as_ref();
-    let dest = dest.as_ref();
+// fn ropycopy<P1: AsRef<Path>, P2: AsRef<Path>>(source: &P1, dest: &P2) -> Result<()> {
+//     let source = source.as_ref();
+//     let dest = dest.as_ref();
 
-    // robocopy C:\source\something.new C:\destination\something /MIR /ZB /W:5 /R:5 /MT:8 /LOG:C:\logs\copy_log.txt
-    let cmd = std::process::Command::new("robocopy")
-        .arg(source)
-        .arg(dest)
-        .arg("/MIR")
-        .arg("/IS")
-        .arg("/W:1")
-        .arg("/R:5")
-        .arg("/MT:2")
-        .output()?;
+//     // robocopy C:\source\something.new C:\destination\something /MIR /ZB /W:5 /R:5 /MT:8 /LOG:C:\logs\copy_log.txt
+//     let cmd = std::process::Command::new("robocopy")
+//         .arg(source)
+//         .arg(dest)
+//         .arg("/MIR")
+//         .arg("/IS")
+//         .arg("/W:1")
+//         .arg("/R:5")
+//         .arg("/MT:2")
+//         .output()?;
 
-    let stdout = String::from_utf8_lossy(&cmd.stdout);
-    let stderr = String::from_utf8_lossy(&cmd.stderr);
+//     let stdout = String::from_utf8_lossy(&cmd.stdout);
+//     let stderr = String::from_utf8_lossy(&cmd.stderr);
 
-    let exit_code = cmd.status.code().unwrap_or(-9999);
-    if (0..7).contains(&exit_code) {
-        info!("{stdout}");
-        info!("Robocopy completed successfully.");
-    } else {
-        error!("{stdout}");
-        error!("{stderr}");
-        bail!("Robocopy failed with code: {:?}", exit_code);
-    }
-    Ok(())
-}
+//     let exit_code = cmd.status.code().unwrap_or(-9999);
+//     if (0..7).contains(&exit_code) {
+//         info!("{stdout}");
+//         info!("Robocopy completed successfully.");
+//     } else {
+//         error!("{stdout}");
+//         error!("{stderr}");
+//         bail!("Robocopy failed with code: {:?}", exit_code);
+//     }
+//     Ok(())
+// }
 
 pub fn apply_package_impl(old_locator: &VelopackLocator, package: &PathBuf, run_hooks: bool) -> Result<VelopackLocator> {
     let mut bundle = load_bundle_from_file(package)?;
@@ -87,49 +87,55 @@ pub fn apply_package_impl(old_locator: &VelopackLocator, package: &PathBuf, run_
 
         // third, we try _REALLY HARD_ to stop the package
         let _ = shared::force_stop_package(root_path);
-        if winsafe::IsWindows10OrGreater() == Ok(true) && !locksmith::close_processes_locking_dir(&old_locator) {
-            bail!("Failed to close processes locking directory / user cancelled.");
-        }
+        // if winsafe::IsWindows10OrGreater() == Ok(true) && !locksmith::close_processes_locking_dir(&old_locator) {
+        //     bail!("Failed to close processes locking directory / user cancelled.");
+        // }
 
         // fourth, we make as backup of the current dir to temp_path_old
         info!("Backing up current dir to {}", &temp_path_old.to_string_lossy());
-        let mut requires_robocopy = false;
-        if let Err(e) = fs::rename(&current_dir, &temp_path_old) {
-            warn!("Failed to rename current_dir to temp_path_old ({}). Retrying with robocopy...", e);
-            ropycopy(&current_dir, &temp_path_old)?;
-            requires_robocopy = true;
-        }
+        shared::retry_io_ex(|| fs::rename(&current_dir, &temp_path_old), 1000, 10)
+            .context("Unable to start the update, because one or more running processes prevented it.")?;
+
+        // let mut requires_robocopy = false;
+        // if let Err(e) = fs::rename(&current_dir, &temp_path_old) {
+        //     warn!("Failed to rename current_dir to temp_path_old ({}). Retrying with robocopy...", e);
+        //     ropycopy(&current_dir, &temp_path_old)?;
+        //     requires_robocopy = true;
+        // }
 
         // fifth, we try to replace the current dir with temp_path_new
         // if this fails we will yolo a rollback...
         info!("Replacing current dir with {}", &temp_path_new.to_string_lossy());
+        shared::retry_io_ex(|| fs::rename(&temp_path_new, &current_dir), 1000, 30)
+            .context("Unable to complete the update, and the app was left in a broken state. You may need to re-install")?;
 
-        if !requires_robocopy {
-            // if we didn't need robocopy for the backup, we don't need it for the deploy hopefully
-            if let Err(e1) = fs::rename(&temp_path_new, &current_dir) {
-                warn!("Failed to rename temp_path_new to current_dir ({}). Retrying with robocopy...", e1);
-                requires_robocopy = true;
-            }
-        }
+        // if !requires_robocopy {
+        //     // if we didn't need robocopy for the backup, we don't need it for the deploy hopefully
+        //     if let Err(e1) = fs::rename(&temp_path_new, &current_dir) {
+        //         warn!("Failed to rename temp_path_new to current_dir ({}). Retrying with robocopy...", e1);
+        //         requires_robocopy = true;
+        //     }
+        // }
 
-        if requires_robocopy {
-            if let Err(e2) = ropycopy(&temp_path_new, &current_dir) {
-                error!("Failed to robocopy temp_path_new to current_dir ({}). Will attempt a rollback...", e2);
-                let _ = ropycopy(&temp_path_old, &current_dir);
-                let _ = tx.send(splash::MSG_CLOSE);
+        // if requires_robocopy {
+        //     if let Err(e2) = ropycopy(&temp_path_new, &current_dir) {
+        //         error!("Failed to robocopy temp_path_new to current_dir ({}). Will attempt a rollback...", e2);
+        //         let _ = ropycopy(&temp_path_old, &current_dir);
+        //         let _ = tx.send(splash::MSG_CLOSE);
 
-                info!("Showing error dialog...");
-                let title = format!("{} Update", &new_locator.get_manifest_title());
-                let header = "Failed to update";
-                let body =
-                    format!("Failed to update {} to version {}. Please check the logs for more details.",
-                            &new_locator.get_manifest_title(),
-                            &new_locator.get_manifest_version_full_string());
-                dialogs::show_error(&title, Some(header), &body);
+        //         info!("Showing error dialog...");
+        //         let title = format!("{} Update", &new_locator.get_manifest_title());
+        //         let header = "Failed to update";
+        //         let body = format!(
+        //             "Failed to update {} to version {}. Please check the logs for more details.",
+        //             &new_locator.get_manifest_title(),
+        //             &new_locator.get_manifest_version_full_string()
+        //         );
+        //         dialogs::show_error(&title, Some(header), &body);
 
-                bail!("Fatal error performing update.");
-            }
-        }
+        //         bail!("Fatal error performing update.");
+        //     }
+        // }
 
         // from this point on, we're past the point of no return and should not bail
         // sixth, we write the uninstall entry
@@ -146,7 +152,7 @@ pub fn apply_package_impl(old_locator: &VelopackLocator, package: &PathBuf, run_
         } else {
             info!("Skipping uninstall entry for portable app.");
         }
-      
+
         // seventh, we run the post-install hooks
         if run_hooks {
             crate::windows::run_hook(&new_locator, constants::HOOK_CLI_UPDATED, 15);

--- a/src/bins/src/lib.rs
+++ b/src/bins/src/lib.rs
@@ -4,7 +4,7 @@ pub mod shared;
 #[cfg(target_os = "windows")]
 pub mod windows;
 
-pub use shared::{dialogs};
+pub use shared::dialogs;
 
 #[macro_use]
 extern crate log;

--- a/src/bins/src/shared/util_common.rs
+++ b/src/bins/src/shared/util_common.rs
@@ -29,39 +29,25 @@ where
     F: Fn() -> Result<T, E>,
     E: std::fmt::Debug,
 {
-    let res = op();
-    if res.is_ok() {
-        return Ok(res.unwrap());
+    retry_io_ex(op, 1000, 4)
+}
+
+pub fn retry_io_ex<F, T, E>(op: F, delay_ms: i32, count: i32) -> Result<T, E>
+where
+    F: Fn() -> Result<T, E>,
+    E: std::fmt::Debug,
+{
+    let mut res = op();
+    for _ in 0..count {
+        if res.is_ok() {
+            return Ok(res.unwrap());
+        }
+
+        warn!("Retrying operation in {}ms... (error was: {:?})", delay_ms, res.err());
+        thread::sleep(Duration::from_millis(delay_ms as u64));
+        res = op();
     }
-
-    warn!("Retrying operation in 1000ms... (error was: {:?})", res.err());
-    thread::sleep(Duration::from_millis(1000));
-
-    let res = op();
-    if res.is_ok() {
-        return Ok(res.unwrap());
-    }
-
-    warn!("Retrying operation in 1000ms... (error was: {:?})", res.err());
-    thread::sleep(Duration::from_millis(1000));
-
-    let res = op();
-    if res.is_ok() {
-        return Ok(res.unwrap());
-    }
-
-    warn!("Retrying operation in 1000ms... (error was: {:?})", res.err());
-    thread::sleep(Duration::from_millis(1000));
-
-    let res = op();
-    if res.is_ok() {
-        return Ok(res.unwrap());
-    }
-
-    warn!("Last retry in 1000ms... (error was: {:?})", res.err());
-    thread::sleep(Duration::from_millis(1000));
-
-    op()
+    res
 }
 
 pub fn random_string(len: usize) -> String {

--- a/src/bins/src/windows/mod.rs
+++ b/src/bins/src/windows/mod.rs
@@ -1,4 +1,4 @@
-pub mod locksmith;
+// pub mod locksmith;
 pub mod mitigate;
 pub mod prerequisite;
 pub mod runtimes;

